### ISSUE TITLE
Match reporting error 

### ISSUE THREAD

**Issue by charles-paperman** [V0ldek/rsonpath#247]
### Package

Library

### Describe the bug

When using RsonpathEngin.matches  the end offset is pointing toward the next structural character.

### Minimal Reproducible Example

With the JSON

```json
{"a": {"b": "c"}, "c": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]}'
```

* The query `.a.b` get the matches 12-16 which gives the slice:  `'"c"}'`
* The query `.c[5]` get the matches 39-41 which gives the slice:  `5,`

### Expected behavior

With the JSON

```json
{"a": {"b": "c"}, "c": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]}'
```

* The query `.a.b` get the matches 12-15 which gives the slice:  `"c"`
* The query `.c[5]` get the matches 39-41 which gives the slice:  `5`

### Version of the release

v0.7.0

### Rust version

nightly

### Target triple

x86_64

### Features enabled

default

**Comment by V0ldek** [comment]
Yup, full repro.

The bug is in the `nodes` recorders, both the simple one ([line 308](https://github.com/V0ldek/rsonpath/blob/515eb4ec15910e6d289cb39f008e63c0556f7895/crates/rsonpath-lib/src/result/nodes.rs#L308) and the stack ([line 475](https://github.com/V0ldek/rsonpath/blob/515eb4ec15910e6d289cb39f008e63c0556f7895/crates/rsonpath-lib/src/result/nodes.rs#L475)). We `finalize_node` to trim the whitespace but don't actually propagate that to the end index. The fix is easy, the end index should be simply taken as `start_idx + buf.len() - 1`.

Will prioritize for next version.

**Comment by V0ldek** [comment]
Fixed in v0.8.1
